weldx/geometry.py

```diff
@@ -99,255 +98,404 @@ def _to_list(var) -> list:
 
-    def apply_transformation(self, matrix):
-        """Apply a transformation matrix to the segment.
+    def apply_transformation(self, matrix: np.ndarray) -> DynamicShapeSegment:
+        """Apply an in-place transformation to the segment using a matrix.
 
         Parameters
         ----------
-        matrix :
-            Transformation matrix
+        matrix:
+            The transformation matrix
+
+        Returns
+        -------
+        DynamicShapeSegment:
+            A self-reference to the modified segment
 
         """
-        self._points = np.matmul(matrix, self._points)
-        self._calculate_length()
+        if self._series.is_expression:
+            raise NotImplementedError
 
-    @UREG.wraps(None, (None, _DEFAULT_LEN_UNIT), strict=True)
-    def apply_translation(self, vector: pint.Quantity):
-        """Apply a translation to the segment.
+        matrix_33 = np.eye(3)
+        matrix_33[:2, :2] = matrix
+        matrix_33 = ut.xr_3d_matrix(matrix_33)
+
+        dat = ut.xr_matmul(
+            matrix_33, self._series.data_array, dims_a=["c", "v"], dims_b=["c"]
+        )
+        self._series = SpatialSeries(
+            dat.transpose(..., "c").data, coords={"s": dat.coords["s"].data}
+        )
+        self._length = self.get_section_length(self._max_coord)
+        return self
+
+    def transform(self, matrix):
+        """Create a new segment transformed by the passed matrix.
 
         Parameters
         ----------
-        vector :
-            Translation vector
+        matrix:
+            The transformation matrix
+
+        Returns
+        -------
+        DynamicShapeSegment:
+            The transformed segment
 
         """
-        self._points += np.ndarray((2, 1), float, np.array(vector, float))
+        new_segment = copy.deepcopy(self)
+        return new_segment.apply_transformation(matrix)
 
-    @UREG.wraps(_DEFAULT_LEN_UNIT, (None, _DEFAULT_LEN_UNIT), strict=True)
-    def rasterize(self, raster_width: pint.Quantity) -> pint.Quantity:
-        """Create an array of points that describe the segments contour.
 
-        The effective raster width may vary from the specified one,
-        since the algorithm enforces constant distances between two
-        raster points.
+# LineSegment -----------------------------------------------------------------
+
+
+class LineSegment(DynamicShapeSegment):
+    """Line segment."""
+
+    @UREG.wraps(None, (None, _DEFAULT_LEN_UNIT), strict=True)
+    def __init__(self, points: pint.Quantity):
+        """Construct line segment.
 
         Parameters
         ----------
-        raster_width :
-            The desired distance between two raster points
+        points :
+            2x2 matrix of points. The first column is the
+            starting point and the second column the end point.
 
         Returns
         -------
-        pint.Quantity
-            Array of contour points
+        LineSegment
```

Comment:
A short example in the `__init__` docstring here would be nice to avoid confusion when creating the 2x2 point matrix

Reply:
done